pantheon-bundle/frontend/src/app/product.tsx

```diff
@@ -94,8 +98,24 @@ class Product extends Component<any, IState> {
                                     <TextInput id="product_description_text" type="text" placeholder="Product Description" value={this.state.productDescription} onChange={this.handleProductInput} />
                                 </FormGroup>
                                 <br />
+                                <FormGroup
+                                        label="Product Version:"
+                                        isRequired={true}
+                                        fieldId="version_name"
+                                >
+                                    <TextInput id="new_version_name_text" type="text" placeholder="Product Version" onChange={this.handleTextInputChange} value={this.state.versionName} />
+                                </FormGroup>
+                                <br />
+                                <FormGroup
+                                        label="Version URL Fragment:"
+                                        isRequired={true}
+                                        fieldId="version_url_fragment"
+                                >
+                                    <TextInput id="new_version_url_fragment" type="text" placeholder="Version URL Fragment" onChange={this.handleUrlInputChange} value={this.state.versionUrlFragment} />
+                                </FormGroup>
+                                <br />
                                 <ActionGroup>
-                                    <Button aria-label="Creates a new Product Name with Description specified." onClick={this.saveProduct}>Save</Button>
+                                    <Button type='submit' aria-label="Creates a new Product Name with Description specified." onClick={e => this.saveProduct(e)}>Save</Button>
```

Comment:
we don't need the lambda function for onClick event or the Button type. I recommend:

<Button aria-label="Creates a new Product Name with Description specified." onClick={this.saveProduct}>Save</Button>